flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobVertexTaskManagersInfo.java

```diff
@@ -185,6 +192,7 @@ public boolean equals(Object o) {
         public int hashCode() {
             return Objects.hash(
                     host,
+                    endpoint,
```

Comment:
I assume this means that the new object will next match an old object - is this intentional? Do we want to check if the object has a location and look to match on hostname and port with the endpoints hostname and port?